### PR TITLE
Victoria Garcia - Ampers - Hotel -updated

### PR DESCRIPTION
**Pull request by Lasiorhine**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | 

There was a question of whether to include a "room" class, and I know that in class, there was lots of discussion about whether it was really necessary.  I chose to make one, because I wanted each room to be the organizer for its own reservations, price, discounts, and availability.  

Ultimately, however, this choice may have complicated things unnecessarily. No, not 'may', probably.  It probably made this a lot more complicated than it needed to be.  Because yiiiiiiii, did this turned out to be complicated.

Describe a concept that you gained more clarity on as you worked on this assignment. | 

So very many.  Array methods, hash methods, chained enumerables and basic loops-- I even got some practice using basic object methods, like is_a?.  I also got a lot more facility with debugging.  I'm pretty handy with pry now, along with various techniques involving commenting-out and interstitial 'puts' statements, that help you get inside a complex method and figure out what's broken. 

Describe a _nominal test_ that you wrote for this assignment. | 

I did a lot of very basic tests for initialize methods:  Does the thing initialize, does it have the variables you want, etc. 

Describe an _edge case test_ that you wrote for this assignment. | 

In the system for tracking a room's unavailable days, I wrote an awful lot of them.  I tested what would happen if a proposed reservation overlapped another, abutted two but did not overlap, etc. I should confess, however, that as I made the final push to finish Wave 3, a lot of the extensive edge-case testing kind of went out the window. 

How do you feel you did in writing pseudocode first, then writing the tests and then the code? | 

It's a really good tool, especially when things seem overwhelming.  I found it little unwieldy during times when I knew what I was doing, however.

### CODE REVIEW

File: lib/front_desk.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      room_array = []
+      for i in 1 .. TOTAL_ROOMS_IN_FACILITY
+        room_number = i.to_s
+        new_room = Hotel::Room.new(room_number)
+        room_array << new_room
+      end
+      return room_array
+    end
+
+    def report_all_rooms
```
Comment (kariabancroft) [inline lib/front_desk.rb]: Why would you need a separate method for this rather than using an `attr`?

File: lib/front_desk.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      return room_to_assign
+    end
+
+    def look_up_per_night_price_for_room(query_room_numb)
+      target_room = @rooms.find {|room| room.room_number == query_room_numb}
+      target_price = target_room.rate_with_discount
+    end
+
+    def locate_room_by_id(query_rm_numb)
+      target_room = @rooms.find {|room| room.room_number == query_rm_numb}
```
Comment (kariabancroft) [inline lib/front_desk.rb]: This logic is duplicated in the method above, so why not just use this instead of duplicating?

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility |Yes - though it ended up getting a bit muddled throughout, I think you did OK overall with the single purpose 
Demonstrated loose coupling |Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Mostly - you seem 50/50 on return values and using those return values when calling the methods
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements |Incomplete

Though I think you may have gotten a bit muddled down in the code overall, I think you hit most of the major points. Simple code is better - so sometimes you may just need to remind yourself of that.

**Comment by Lasiorhine** [comment]
This is the updated pull request for "Hotel Revisited."
